app/bucketlist/views.py

```diff
@@ -0,0 +1,130 @@
+from app.models import User, BucketList, Item
+from flask import request, jsonify, Blueprint
+from app import db, jwt
+from flask_jwt import jwt_required, current_identity
+from app.auth.validate import validate_json
+from datetime import datetime
+from sqlalchemy.sql import func
+
+
+bucketlists = Blueprint('bucketlists', __name__,
+                        url_prefix='/api/v1/bucketlists')
+
+
+# Get all bucketlists and search bucketlists
+@bucketlists.route('/', methods=['GET'])
+@jwt_required()
+def get_bucketlist():
+    search = request.args.get('q', '')
+    page = request.args.get('page', 1)
+    limit = request.args.get('limit', 20)
+
+    results = BucketList.query.filter_by(user_id=current_identity.id).filter(
+        BucketList.name.ilike(f'%{search}%')).paginate(int(page), int(limit), False)
```

Comment:
Use `try... except` block here to ensure that the `page` and `limit` are integers. If i send in `page` and `limit` as letters, your code will break at this point trying to cast strings.